tests/AppTest/Functional/UserTest.php

```diff
@@ -0,0 +1,390 @@
+        $resetPassword->setHash('test');
+        $resetPassword->setExpires((new \DateTimeImmutable())->add(new \DateInterval('P1D')));
+        $user->addResetPassword($resetPassword);
+        $this->getEntityManager()->persist($resetPassword);
+        $this->getEntityManager()->persist($user);
+        $this->getEntityManager()->flush();
+
+        $mailService = $this->createMock(MailService::class);
+        $this->replaceService(MailService::class, $mailService);
+
+        $response = $this->patch('/account/reset-password/' . $resetPassword->getHash(), [
+            'password' => '654321',
+            'passwordConfirm' => '654321',
+        ]);
+
+        $data = json_decode($response->getBody()->getContents(), true);
+
+        $this->assertResponseBadRequest($response);
+        $this->assertArrayHasKey('error', $data);
+        $this->assertArrayHasKey('messages', $data['error']);
+        $this->assertNotEmpty($data['error']['messages'][0]);
+        $this->assertSame(
+            sprintf(Message::RESET_PASSWORD_USED, $resetPassword->getHash()),
+            $data['error']['messages'][0]
+        );
+    }
+
+    public function testResetPassword()
+    {
+        $user = $this->createUser();
+        $resetPassword = new UserResetPasswordEntity();
+        $resetPassword->setUser($user);
+        $resetPassword->setStatus(UserResetPasswordEntity::STATUS_REQUESTED);
+        $resetPassword->setHash('test');
+        $resetPassword->setExpires((new \DateTimeImmutable())->add(new \DateInterval('P1D')));
+        $user->addResetPassword($resetPassword);
+        $this->getEntityManager()->persist($resetPassword);
+        $this->getEntityManager()->persist($user);
+        $this->getEntityManager()->flush();
+
+        $mailService = $this->createMock(MailService::class);
+        $this->replaceService(MailService::class, $mailService);
+
+        $response = $this->patch('/account/reset-password/' . $resetPassword->getHash(), [
+            'password' => '654321',
+            'passwordConfirm' => '654321',
+        ]);
+
+        $data = json_decode($response->getBody()->getContents(), true);
+
+        $this->assertResponseOk($response);
+        $this->assertArrayHasKey('info', $data);
+        $this->assertArrayHasKey('messages', $data['info']);
+        $this->assertNotEmpty($data['info']['messages'][0]);
+        $this->assertSame(Message::RESET_PASSWORD_OK, $data['info']['messages'][0]);
+    }
+
+    public function testResetPasswordByEmail()
+    {
+        $user = $this->createUser();
+
+        $mailService = $this->createMock(MailService::class);
+        $this->replaceService(MailService::class, $mailService);
+
+        $response = $this->post('/account/reset-password', [
+            'email' => $user->getDetail()->getEmail(),
+        ]);
+
+        $this->assertResponseOk($response);
+        $this->assertCount(1, $user->getResetPasswords());
+    }
+
+//    public function testRegisterAccountDuplicateIdentity()
```

Comment:
Do we still need this once it's commented out?

Reply:
Can a user register itself ?

Reply:
With an existing identifier? Whoever (admin/user) tries to create a new user account using an existing identity should get an error telling them that there already is an account registered with that identifier (which, from what I know, already happens).

Reply:
It was a bug related to the guest permission, a guest didn't had access to create a user. It was fixed now.